shopping_list_frontend/lib/services/api/base.dart

```diff
@@ -1,15 +1,30 @@
 import 'package:http/http.dart' as http;
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:shopping_list_frontend/services/api/connectivity.dart';
+
+import 'package:shopping_list_frontend/cubits/general/internet_connection_cubit.dart';
 
 String apiIp = "10.0.2.2";
 
 abstract class BaseRequestSender {
-  Future<Map<String, String>> getTokenHeader() async {
+  ConnectivityCheck connectivityCheck = ConnectivityCheck(onError: () {});
+  InternetConnectionCubit internetConnectionCubit = InternetConnectionCubit();
```

Comment:
please use provider for that staff. to keep one instance per app(i didn't find case when you need 2 or more separate), pls fix it